### ISSUE TITLE
IDEA 2020.3.1 on Gnome: Theme doesn't update automatically while running

### ISSUE THREAD

**Issue by lunaryorn** [weisJ/auto-dark-mode#27]
With IntelliJ 2020.3.1 on Gnome the plugin only changes the theme on startup or after restart, but while IDE is running.  Is this intended?

**Comment by weisJ** [comment]
This certainly isn't supposed to be happening. I will have a look at what might be causing this.

**Comment by lunaryorn** [comment]
Can I do anything to debug it?

**Comment by weisJ** [comment]
It would be very helpful if you could install [this version](https://github.com/weisJ/auto-dark-mode/releases/tag/v1.5.3-203.alpha1) and provide the logs for when changing the gnome theme.

**Comment by lunaryorn** [comment]
Where do I find the logs?  Sorry if that's a stupid question, but I have never debugged am Idea plugin before.

**Comment by weisJ** [comment]
Not a stupid question at all. You can find the logs under Help->Show logs

**Comment by lunaryorn** [comment]
There's an `idea.log` file but it shows no message if I `tail -f` it and change the theme via `gsettings set`.  The only messages related to this plugin I can see are:

```
[…] INFO - tform.linux.gnome.GnomeLibrary - Loaded libauto-dark-mode-linux-gnome.so. 
[…] INFO - .platform.AbstractThemeMonitor - Started theme monitoring. 
```

Haven't noticed any exception :shrug:

**Comment by lunaryorn** [comment]
PS: As an additional data point, the theme updates also if I toggle the "Change editor/IDE theme" check boxes in the settings dialog of this plugin and press "Apply"… 

It looks as if it perfectly picks up the *current* theme, but fails to get an event if the theme's changed.

**Comment by weisJ** [comment]
Yes it seems like the gnome settings hook doesn't fire when the settings change. Strangely enough it seems to be working when using the `runIde` gradle task. Are you using version `2020.3.1` or `2020.3`? There might have been some changes in `2020.3.1` relating to how native libraries are handles in the jetbrains runtime (Which I hope is not the case).

**Comment by lunaryorn** [comment]
2020.3.1

**Comment by weisJ** [comment]
Indeed the signal, that the theme has been changed, doesn't notify the registered callback.
It seems like this behaviour can't be replicated using the `runIde` task (there the theme detection works as expected). This makes it very difficult to debug the issue as this probably is related to how the plugin classes/libraries are loaded in the Jetbrains runtime (This is just speculation. The `runIde` task also used the Jetbrains runtime, so this probably isn't the full picture).
I have submitted a [bug report](https://youtrack.jetbrains.com/issue/IDEA-259945) to Jetbrains, to get some clarification.
